pkg/controller/replica.go

```diff
@@ -300,13 +291,29 @@ func NewReplicaController(mgr manager.Manager, controllerSharedState *SharedStat
 			return false
 		},
 		UpdateFunc: func(e event.UpdateEvent) bool {
-			return true
+			objOld, okOld := e.ObjectOld.(*azdiskv1beta2.AzVolumeAttachment)
+			objNew, okNew := e.ObjectNew.(*azdiskv1beta2.AzVolumeAttachment)
+
+			demoteUpdate := !isDemotionRequested(objOld) && isDemotionRequested(objNew)
+			attachFailedUpdate := objOld.Status.State != azdiskv1beta2.AttachmentFailed && objNew.Status.State == azdiskv1beta2.AttachmentFailed
+			detachFailedUpdate := objOld.Status.State != azdiskv1beta2.DetachmentFailed && objNew.Status.State == azdiskv1beta2.DetachmentFailed
+
+			deleteRequestedOld, _ := objectDeletionRequested(objOld)
+			deleteRequestedNew, _ := objectDeletionRequested(objNew)
+			deleteTimestampUpdate := !deleteRequestedOld && deleteRequestedNew
+
+			updateEventsForReplica := demoteUpdate || attachFailedUpdate || deleteTimestampUpdate || detachFailedUpdate
```

Comment:
Consider adding `objNew.Spec.RequestedRole == azdiskv1beta2.ReplicaRole && (demoteUpdate || attachFailedUpdate || deleteTimestampUpdate || detachFailedUpdate))` to make distinction between update events to be watched for _replicas_ vs. _primaries_.

Reply:
We would need to check for a detach requested event. Otherwise, we wouldn't never be able to trigger a `go replicaHandleDelete` for detach operation.

Reply:
We would trigger `go replicaHandleDelete` with `deleteTimestampUpdate` after the completion of a detach operation.

Reply:
Right 👍